ballerina/modules/soap11/soap11.bal

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import soap;
+import soap.wssec;
+
+import ballerina/http;
+import ballerina/mime;
+
+# Object for the basic SOAP client endpoint.
+public isolated client class Client {
+    private final http:Client soapClient;
+    private final readonly & wssec:InboundSecurityConfig|wssec:InboundSecurityConfig[] inboundSecurity;
+    private final readonly & wssec:OutboundSecurityConfig outboundSecurity;
+
+    # Gets invoked during object initialization.
+    #
+    # + url - URL endpoint
+    # + config - Configurations for SOAP client
+    # + return - `error` in case of errors or `()` otherwise
+    public isolated function init(string url, *soap:ClientConfig config) returns Error? {
+        do {
+            check soap:validateTransportBindingPolicy(config);
+            self.soapClient = check new (url, config.httpConfig);
+            readonly & soap:ClientConfig readonlyConfig = soap:getReadOnlyClientConfig(config);
+            self.inboundSecurity = readonlyConfig.inboundSecurity;
+            self.outboundSecurity = readonlyConfig.outboundSecurity;
+        } on fail var err {
+            return error Error(SOAP_CLIENT_ERROR, err);
+        }
+    }
+
+    # Sends SOAP request and expects a response.
+    # ```ballerina
+    # xml|mime:Entity[] response = check soapClient->sendReceive(body, action);
+    # ```
+    #
+    # + body - SOAP request body as an `XML` or `mime:Entity[]` to work with SOAP attachments
+    # + action - SOAP action as a `string`
+    # + headers - SOAP headers as a `map<string|string[]>`
+    # + path - The resource path
+    # + return - If successful, returns the response. Else, returns an error
+    remote isolated function sendReceive(xml|mime:Entity[] body, string action,
+                                         map<string|string[]> headers = {}, string path = "") returns xml|mime:Entity[]|Error {
+        do {
+            xml securedBody;
+            xml response;
```

Comment:
let's move this after the lock statement

Reply:
Fixed in [902844f](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/902844feeea46ae121bac4d6f4127613ef2cc750)